src/UserService.Data/UserRepository.cs

```diff
@@ -204,11 +204,12 @@ public bool IsExistUser(Guid userId)
             return _provider.Users.FirstOrDefault(u => u.Id == userId) != null;
         }
 
-        public IEnumerable<DbUser> Search(string text)
+        public List<DbUser> Search(string text)
         {
-            return _provider.Users.Where(u => u.LastName.Contains(text, StringComparison.OrdinalIgnoreCase)
-                                            || u.FirstName.Contains(text, StringComparison.OrdinalIgnoreCase)
-                                            || u.MiddleName.Contains(text, StringComparison.OrdinalIgnoreCase));
+            return _provider.Users
+                                .ToList()
```

Comment:
It looks strange cast to list two times? Is it really doesn't work without it?

Reply:
first to list get all users, it is not good, but the condition that I want cannot be fulfilled without it

Reply:
how I understand it, this request can't convert to sql request